packages/translated-components/src/index.js

```diff
@@ -61,16 +62,29 @@ const moneyFormat = (language) => ({
 
 const preHeat = (translations, format) => (
   reduce(translations, (acc, o, language) => {
-    acc[language] = reduce(o, (acc, v, k) => {
-      acc[k] = new IntlFormat(v, kebabCase(language), {...moneyFormat(language), ...format})
-      return acc
-    }, {})
+    acc[language] = preheatValues(o, format, language)
     return acc
   }, {})
 )
 
-const templateReducer = (values) => (acc, v, k) => {
-  acc[k] = v.format(values)
+const preheatValues = (obj, format, language) => {
+  return reduce(obj, (acc, value, key) => {
+    if (typeof value === 'object') {
+      acc[key] = preheatValues(value, format, language)
+    } else {
+      acc[key] = new IntlFormat(value, kebabCase(language), {...moneyFormat(language), ...format})
+    }
+    return acc
+  }, {})
+}
+
+const templateReducer = (values) => partial(templateIteratee, values)
+const templateIteratee = (values, acc, v, k) => {
+  if (typeof v === 'object' && !v.format) {
+    acc[k] = reduce(v, partial(templateIteratee, values), {})
+  } else {
+    acc[k] = v.format(values)
```

Comment:
Might be worth writing a test for the case that a `translations` object is passed in with a `format` key which is a string to be translated.